.github/workflows/docker-build-deploy.yml

```
@@ -48,7 +57,8 @@ jobs:
   docker-build:
     needs: check-version
     runs-on: ubuntu-latest
-    if: needs.check-version.outputs.should-build == 'true' && (github.event_name == 'push' || (github.event_name == 'pull_request' && github.event.pull_request.merged == true))
+    timeout-minutes: 30
+    if: needs.check-version.outputs.should-build == 'true' && (github.event_name == 'push' || (github.event_name == 'pull_request' && github.event.pull_request.merged == true)) && secrets.DOCKER_USERNAME != '' && secrets.DOCKER_PASSWORD != ''
```

actionlint (1.7.7)

context "secrets" is not allowed here. available contexts are "github", "inputs", "needs", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)